### PR TITLE
fix(action-popover-item): prevent popover staying open when using keyboard nav (FE-4703) 

### LINKED ISSUE #4662: ActionPopoverMenuButton - Stay open behind when we open a Dialog
### Current behaviour

When using ActionPopoverMenuButton in a FlatTable with the keyboard, this one stay open when we open a dialog.

### Expected behaviour

ActionPopover would be closed when we open a dialog in front of, like when we use the mouse.

### CodeSandbox or Storybook URL

https://codesandbox.io/s/black-moon-ugf2n?file=/src/index.js

### JIRA Ticket (Sage Only)

SBS-25708

### Suggested Solution

_No response_

### Carbon Version

101.4.2

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

Windows

### Anything else we should know?

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by DipperTheDan**
fix #4662

### Proposed behaviour
action-popover-bug-fix

### Current behaviour
action-popover-bug-original

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

File: src/components/action-popover/action-popover.stories.mdx [hunk 1]
```diff
@@ -82,12 +84,170 @@ import {
 
 ## Examples
 
+## TEST_TEST
```
Comment (DipperTheDan) [inline src/components/action-popover/action-popover.stories.mdx:87]: Don't be alarmed by this story. It's here to locally replicate the code used in the Github Issue. Makes manually testing the bug easier.

**Review by grabkowski — DISMISSED** [review]

File: src/components/action-popover/action-popover.stories.mdx (hunk 1 above)
Comment (edleeks87) [reply src/components/action-popover/action-popover.stories.mdx]: I've moved this story and renamed it as there's some value in having it IMO, I'll catch up with QA about whether cypress tests can be added

**Review by grabkowski — DISMISSED** [review]

File: src/components/action-popover/action-popover-test.stories.mdx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import { useState } from "react";
```
Comment (samtjo) [inline src/components/action-popover/action-popover-test.stories.mdx:1]:
```suggestion
```

File: src/components/action-popover/action-popover.spec.js [hunk 3]
```diff
@@ -259,18 +259,19 @@ describe("ActionPopover", () => {
...
 
       act(() => {
-        wrapper
-          .find(StyledMenuItem)
-          .props()
-          .onKeyDown({ which: 13, stopPropagation: jest.fn() });
+        wrapper.find(StyledMenuItem).props().onKeyDown({
+          which: 13,
+          stopPropagation: jest.fn(),
+          preventDefault: jest.fn(),
```
Comment (samtjo) [inline src/components/action-popover/action-popover.spec.js:273]: No need to make these `jest.fn()` if we're not checking they've been called anywhere
```suggestion
          preventDefault: () => {},
```

**Review by samtjo — CHANGES_REQUESTED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by samtjo — APPROVED** [review]

**Review by divyajindel — APPROVED** [review]

**Pull request merged**